### PR TITLE
Add data validity tests

### PR DESCRIPTION
**Pull request by russellpierce**
* Add regression tests
* Add expected_output for regression tests
* Add suggestion for dplyr and testthat (>= 2.0.0)

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Package: babynames
 Title: US Baby Names 1880-2015
-Version: 0.3.0
+Version: 0.3.1
```
Comment (hadley) [inline DESCRIPTION]: Can you please bump this to the standard dev version? (i.e. 0.3.0.9000)

**Comment by russellpierce** [comment]
Sure I'll make those changes. 

FWIW the covariance table seems like a much stronger equivalency test than the head/tail. But, I don't feel super strongly about it.

For first and last 10 rows, are you content with a head bind_rows() on tail, or do you have a preferred idiom?

File: R/data.R [hunk 2]
```diff
@@ -53,3 +53,23 @@ NULL
...
+
+#' Return the head and tail of a data.freame
+#'
+#' For the purposes of DRY testing only.  Do not export.
+#'
+#' @param x
+#'
+#' @return data.frame
+#' @importFrom("utils", "head", "tail")
+first_last <- function(x) {
```
Comment (hadley) [inline R/data.R]: This shouldn't be exported, and can just live in one of the test files.

I think you can eliminate the use of dplyr with something like:

```R
n <- nrow(x)
x[c(1:10, (n - 10):n)]
```

File: tests/testthat/test-applicants_regression.R [hunk 3]
```diff
@@ -0,0 +1,10 @@
+context("applicants regression")
+max_year <- 2015
+
+test_that(paste0("applicants head and tail between start and ", max_year, " unchanged"), {
```
Comment (hadley) [inline tests/testthat/test-applicants_regression.R]: I don't think you need such a complex test name.

Comment (hadley) [reply tests/testthat/test-applicants_regression.R]: And I think it might make more sense to put all the tests in a single file, and call it `test-data.R`

Comment (hadley) [reply tests/testthat/test-applicants_regression.R]: Also you shouldn't be filtering the max year; we want to see when something changes in the tail of the data.

**Comment by russellpierce** [comment]
Yeah, I completely misunderstood what you wanted tested. I assumed this was a test to make sure the underlying data didn't change / the import didn't get borked.. I'm now given to understand that you want the test failure to really highlight the additional row and mostly assume the historical data won't be changed incorrectly.

File: R/data.R (hunk 2 above)
Comment (hadley) [reply R/data.R]: Oh I was confused by the documentation

**Comment by russellpierce** [comment]
Note 1: Because of https://github.com/hadley/babynames/issues/26 and/or devtools/testthat/how the data is currently stored, {tibble} gets attached during the testing of the package.  This makes all of the tests dependent on the behavior of tibble.  

Note 2:  In order to avoid going down a rabbit hole I didn't sort the incoming data prior to applying the test.  Currently the sort order of the individual data sets aren't always consistent.  I've added a note about this in a comment near the top of `test-data.R`.  I'll recommend that when we get around to adding the 2017 data that we enforce sort order at dataset creation time.

**Comment by hadley** [comment]
Sounds good - thanks!

**Pull request merged**